scripts/interactive_tests/interactive_test_util.js

```diff
@@ -62,15 +72,20 @@ var Client = function(port) {
   // Send a command to the server and wait for a response. Return response as a 
   // promise. 
   this.sendCommand = function(cmd) {
-    var deferred = q.defer();
+    let deferredResolve;
```

Comment:
Thanks for the thorough and thoughtful response!

1. This is totally fair. Agreed that it is sad. I'm fine with moving forward a half-step by implementing things natively and then trying to transition control flow over (if possible) in the future.

2. Interesting; I think I still feel that we would be better served by moving this into a helper because we creating the same thing as `q.defer` regardless. If you feel strongly that the "weight" of having to copy the pattern is discourages others from repeating it then I'm fine with keeping things the way they are. Either way, I like your idea of calling out sites where we may run into issues with control flow.